### PR TITLE
Initial very simple skeleton of bot.

### PR DESCRIPTION
**Pull request by egnor**
Basic README and do-nothing functionality (with some logging).

### CODE REVIEW

File: escape_roomba/context.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        self._event_listeners = {}
+
+    def add_listener(self, event_name, handler):
+        """Registers an async function for a Discord event (like 'on_message').
+        Multiple functions can be registered for the same event name."""
+
+        # discord.Client only supports one (static-ish) callback per event
+        # (discordpy.readthedocs.io/en/latest/api.html#discord.Client.event);
+        # discord.ext.commands.Bot/.Cog allow multiple subscribers, but also
+        # assume a specific UX, so we implement our own simple broadcasting.
```
Comment (anaulin) [inline escape_roomba/context.py]: Nit: Is this comment meant for the whole of `add_listener`? Not immediately clear as I read this code top to bottom. If it is meant for `add_listener`, consider moving the comment just above/before the method starts? Also, I don't know what "static-ish" means in practical terms, so I find it kinda confusing -- either explain what you mean, or remove it?

File: escape_roomba/run_bot.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+class StatusLogger:
+    """Logs notable server/connection events."""
+
+    def __init__(self, context):
+        self.context = context
+        context.add_listener_methods(self)
+
+    async def on_connect(self):
+        logging.info('Connected to Discord:\n'
+                     f'    {self.context.client.ws.gateway}')
```
Comment (anaulin) [inline escape_roomba/run_bot.py:47]: Nit: Is this whitespace intentional? (Maybe it looks good in practice, not sure.)

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+[tool.poetry]
+name = "escape_roomba"
+version = "0.1.0"
+description = ""
+authors = ["Your Name <you@example.com>"]
```
Comment (anaulin) [inline pyproject.toml]: Nit: Add your name!

**Review by anaulin — APPROVED** [review]
Looks good. I also went through the steps and was able to create a bot and connect it to a server. I've added a couple of minor changes to the PR, for clarity (a TOC for the README, an improvement to a log message).

My only thought for improvement is maybe having a comment before/above `add_listener` with the general overview of how this works -- i.e. you can add methods as listeners for events, this `Context` class keeps a list and runs them all in parallel on each event. But that's very minor.

👍 🤖 :shipit:

File: escape_roomba/context.py (hunk 1 above)
Comment (egnor) [reply escape_roomba/context.py]: Moved up, removed the "static-ish" comment (it's not worth trying to explain).

File: escape_roomba/run_bot.py (hunk 2 above)
Comment (egnor) [reply escape_roomba/run_bot.py:47]: It is! Indents in the log output.

File: pyproject.toml (hunk 3 above)
Comment (egnor) [reply pyproject.toml]: Done!

**Pull request merged**